### DISCUSSION TITLE
Evaluate `select` filter like `Jq`

### DISCUSSION THREAD

**Discussion by marcellodesales** [mikefarah/yq#708] [Q&A] [answered]
# `jq` can filter with `select`

* With any Json, I can quickly parse/evaluate the string value...
  * I want to ditch `jq` in this case that I need to parse from `yaml -> json`
 
```console
$ docker-compose config | \
      docker run --rm -i mikefarah/yq e '.' - -j | \
      docker run --rm -i imega/jq -r '..|.image? | select(.!=null)'
super/man:0.1.0-debug
super/man:0.1.0-dependencies
super/man:0.1.0
```

# `yq`with `select`?

* I can see the same result with null values... 
* Here's a simple example that returns an extra null at the beginning
  * Is this a bug?
 
```console
$ echo "a:\n  image: super/man:1\nb:\n  image: super/man:2\nc:\n  other: sisi" | \
    docker run --rm -i mikefarah/yq e '.. | .image' -
null
super/man:1
super/man:2
null
```

* Trying to use the filter: it returns the prefix with "!" as if it were the key

```console
$ echo "a:\n  image: super/man:1\nb:\n  image: super/man:2\nc:\n  other: sisi" | \
    docker run --rm -i mikefarah/yq e '.. | .image | select(.!=null)' -
'!': null
super/man:1
super/man:2
'!': null
```

# Question: How can I filter with select or any other function available?

* How to filter elements in any key???

**Comment by mikefarah** [comment, marked as answer]
Ah I haven't (yet) added a `!=` operator, it's on the list. Until then, you can use the not operator:

```
yq e '.. | .image | select( (. == null) | not)' -
```

**Reply by marcellodesales** [reply]
@mikefarah Perfect! 

```console
$ echo "a:\n  image: super/man:1\nb:\n  image: super/man:2\nc:\n  other: sisi" | \
    docker run --rm -i mikefarah/yq e '.. | .image | select((.==null) | not)' -
super/man:1
super/man:2
```
